tests/integration_tests/performance/test_network_latency.py

```diff
@@ -0,0 +1,202 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+"""Tests the network latency of a Firecracker guest."""
+
+import logging
+import platform
+import re
+import pytest
+import host_tools.network as net_tools
+from conftest import _test_images_s3_bucket
+from framework.artifacts import ArtifactCollection, ArtifactSet
+from framework.matrix import TestMatrix, TestContext
+from framework.builder import MicrovmBuilder
+from framework.statistics import core, consumer, producer
+from framework.utils import eager_map
+
+
+PING = "ping -c {} -i {} {}"
+BASELINES = {
+    "x86_64": {
+        "target": 0.140,
+        "delta": 0.025
+    },
+    "aarch64": {
+        "target": 0.110,
+        "delta": 0.015
+    }
+}
+
+INT_REG = r'[0-9]+'
+FLOAT_REG = r'[+-]?[0-9]+\.[0-9]+'
+PKT_LOSS = "pkt_loss"
+LATENCY = "latency"
+
+
+def pass_criteria():
+    """Define pass criteria for the statistics."""
+    # Auxiliary data.
+    delta = BASELINES[platform.machine()]["delta"]
+    target = BASELINES[platform.machine()]["target"]
+
+    # Add a pass criteria for AVG.
+    criteria = dict()
+    criteria[consumer.StatisticDef.AVG_KEY] = consumer.Criteria.eq(target,
+                                                                   delta)
+    return criteria
+
+
+def measurements():
+    """Define the produced measurements."""
+    ms_defs = [consumer.MeasurementDef(LATENCY, "millisecond"),
+               consumer.MeasurementDef(PKT_LOSS, "percentage")]
+    return ms_defs
+
+
+def stats():
+    """Define statistics based on the measurements."""
+    criteria = pass_criteria()
+    # Add default statistics for "latency" measurement.
+    stats_defs = consumer.StatisticDef.defaults(LATENCY,
+                                                pass_criteria=criteria)
+    stats_defs.append(consumer.StatisticDef(PKT_LOSS, PKT_LOSS,
+                                            consumer.Method.last))
+    return stats_defs
+
+
+def consume_ping_output(cons, raw_data, requests):
+    """Consume ping output.
+
+    Output example:
+    PING 8.8.8.8 (8.8.8.8) 56(84) bytes of data.
+    64 bytes from 8.8.8.8: icmp_seq=1 ttl=118 time=17.7 ms
+    64 bytes from 8.8.8.8: icmp_seq=2 ttl=118 time=17.7 ms
+    64 bytes from 8.8.8.8: icmp_seq=3 ttl=118 time=17.4 ms
+    64 bytes from 8.8.8.8: icmp_seq=4 ttl=118 time=17.8 ms
+
+    --- 8.8.8.8 ping statistics ---
+    4 packets transmitted, 4 received, 0% packet loss, time 3005ms
+    rtt min/avg/max/mdev = 17.478/17.705/17.808/0.210 ms
+    """
+    eager_map(cons.consume_measurement_def, measurements())
```

Comment:
nit: might as well be:

```
[cons.consume_measurement_def(i) for i in measurements()]
[cons.consume_stat_def(i) for i in stats()]
```

Reply:
This results in python styling issues, since the result of the list comprehension is not assigned/used by anything, so we would need to discard by assigning them to `don't care ( _ )`. I prefer an `eager_map` approach instead.